### PR TITLE
Add publisher script

### PR DESCRIPTION
**Pull request by btashton**
Running this script will generate the static website content and make a commit in the asf-site branch that is linking to the commit that it generated from.  This also takes care of installing all the ruby build dependencies in a vendor folder so they do not have to be install system wide.

I have an example of a orphaned asf-site branch as well that I would recommend we use as a starting branch.  Someone would need to force it over the existing asf-site branch that is based on master.

So in short to see how this works, clone my fork checkout this publish branch and run the "./publish.sh" script.

### CODE REVIEW

**Comment by huahang** [comment]
LGTM

**Comment by huahang** [comment]
@btashton Hi Brennan, I tried the `publish.sh` script. I think the link to the original version is very nice~
:-)

One thing I've noticed is that this script will put the generated files inside a `content` folder like the Apache Daffodil project [1]. This is different from the Apache brpc project which places the files in `asf-site`'s root folder. Maybe both are supported. We may want to have a try after your other 2 PRs are merged.
:-P

[1]. https://github.com/apache/incubator-daffodil-site/tree/asf-site
[2]. https://github.com/apache/incubator-brpc-website/tree/asf-site

**Comment by xiaoxiang781216** [comment]
> I have an example of a orphaned asf-site branch as well that I would recommend we use as a starting branch. Someone would need to force it over the existing asf-site branch that is based on master.
> 

@btashton I just recreate a ampty asf-site branch, please try it.

**Comment by btashton** [comment]
I just tweaked it to include the asf.yaml
@huahang can you use the publish script to create a PR against it now? (Probably need to hard reset your local asf-site branch)

**Comment by justinmclean** [comment]
Hi,
> One thing I've noticed is that this script will put the generated files inside a content folder like the Apache Daffodil project [1]. This is different from the Apache brpc project which places the files in asf-site's root folder. Maybe both are supported. We may want to have a try after your other 2 PRs are merged.
> :-P
> 

Both are supported, but the content folder being special may go away at some point in the future. (In fact if you make a directory anywhere called content it will replace your web site, I found that out the hard way a few months back.)

Thanks,
Justin

**Comment by btashton** [comment]
Its much more convenient to have a specific folder to add during the git commit to avoid committing build artifacts (.sass-cache and vendor are created), so until that changes I would prefer to keep it as-is.
